EIPS/eip-3156.md

```diff
@@ -255,12 +240,18 @@ contract FlashLender is IERC3156FlashLender {
     }
 
     /**
-     * @dev The amount of currency available to be lended.
+     * @dev Loan `amount` tokens to `receiver`, and takes it back plus a `flashFee` after the callback.
+     * @param receiver The contract receiving the tokens, needs to implement the `onFlashLoan(address user, uint256 amount, uint256 fee, bytes calldata)` interface.
      * @param token The loan currency.
-     * @return The amount of `token` that can be borrowed.
+     * @param amount The amount of tokens lent.
+     * @param data A data parameter to be passed on to the `receiver` for any custom use.
      */
-    function flashSupply(address token) external view override returns (uint256) {
-        return supportedTokens[token] ? IERC20(token).balanceOf(address(this)) : 0;
+    function flashLoan(address receiver, address token, uint256 amount, bytes calldata data) external override {
+        require(supportedTokens[token], "FlashLender: Unsupported currency");
+        uint256 fee = _flashFee(token, amount);
+        IERC20(token).transfer(receiver, amount);
+        IERC3156FlashBorrower(receiver).onFlashLoan(msg.sender, token, amount, fee, data);
+        IERC20(token).transferFrom(receiver, address(this), amount.add(fee));
```

Comment:
`transferFrom` doesn't necessarily throw on failure.  Sadly, some tokens don't return `true` on success either.  ERC20 spec says you **MUST** return `true` on success, so if you want to be spec compliant, wrap this call in a `require(...)`.  Same is true for the `.transfer(...)` call above.

Note: Will do a full review tomorrow, just wanted to call this one out early so you can get a head start on fixing it.

Reply:
Alternatively, check balance before and after the call to make sure that the transfers went through successfully.  Downside is that it is another bunch of gas to do that.

Reply:
Fixed.